docs/src/main/sphinx/security/file-system-access-control.rst

```diff
@@ -331,6 +359,94 @@ The example below defines the following table access policy:
       ]
     }
 
+.. _system-file-function-rules:
+
+Function rules
+^^^^^^^^^^^^^^
+
+These rules control the user's ability to execute SQL functions, including PTFs.
+
+Each function rule is composed of the following fields:
+
+* ``user`` (optional): ``regex`` to match against user name. Defaults to ``.*``.
+* ``role`` (optional): ``regex`` to match against role names. Defaults to ``.*``.
+* ``group`` (optional): ``regex`` to match against group names. Defaults to ``.*``.
+* ``catalog`` (optional): ``regex`` to match against catalog name. Defaults to
+  ``.*``.
+* ``schema`` (optional): ``regex`` to match against schema name. Defaults to ``.*``.
+* ``function`` (optional): ``regex`` to match against function names. Defaults to
+  ``.*``.
+* ``privileges`` (required): zero or more of ``EXECUTE``, ``GRANT_EXECUTE``.
+* ``function_kinds`` (required): one or more of ``AGGREGATE``, ``SCALAR``,
+  ``TABLES``, ``WINDOW``. When a user defines a rule for ``AGGREGATE``, ``SCALAR``
+  or ``WINDOW`` functions, the ``catalog`` and ``schema`` fields are disallowed.
+
+
+To restore previous behavior, which denied all PTFs from any catalog, use the following
+rules:
+
+.. code-block:: json
+
+    {
+      "functions": [
+        {
+          "privileges": [
+            "EXECUTE",
+            "GRANT_EXECUTE"
+          ],
+          "function_kinds": [
+            "SCALAR",
+            "AGGREGATE",
+            "WINDOW"
+          ]
+        },
+        {
+          "privileges": [],
+          "function_kinds": [
+            "TABLE"
+          ]
+        }
+      ]
+    }
+
+
+The following example allows the ``admin`` user to execute ``query`` PTFs from any
+catalog:
+
+.. note::
+
+    It is recommended to only allow admins to use the ``query`` polymorphic table
+    function.
```

Comment:
Add "why"

Maybe it would be nice to have some best-practices somewhere. Where you would have config to grant access to PTF to admin, then create a (definer) view on top of it with admin as owner and granting access to view to regular users. Maybe we could explain that only somewhere.

Reply:
I would even say .. remove that.. there is no reason for this to be best practice in general. Instead explain what the consequence is of this setup .. and earlier what happens by default

Reply:
As in like .. if anyone is allowed to execute table functions.. then what?

Reply:
`query` table function which works as a query pass-though - it's bypass the whole access control so IMO it's quite important to at least mention about it

Reply:
Yeah .. mention.. but not recommend as best practice.. its just important to understand that it is kind of a security hole otherwise ;-) .. so we need to explain that and not just have a recommendation without detailing the reasoning